ssz/sedes/serializable.py

```diff
@@ -132,7 +132,7 @@ def __len__(self):
         return len(self._meta.fields)
 
     def __eq__(self, other):
-        return isinstance(other, BaseSerializable) and hash(self) == hash(other)
+        return self.__class__ == other.__class__ and hash(self) == hash(other)
```

Comment:
*nitpick* since classes are singletons, `self.__class__ is other.__class__` is maybe slightly better here.